platform/ui/src/components/StepProgressDropdown/StepProgressDropdown.tsx

```diff
@@ -0,0 +1,326 @@
+  option: DROPDOWN_OPTION_PROPTYPE.isRequired,
+};
+
+const StepProgressDropdownItem = ({ option, onSelect }) => {
+  const { value } = option;
+
+  return (
+    <div
+      key={value}
+      className={
+        'flex py-1 cursor-pointer hover:bg-secondary-main transition duration-1000'
+      }
+      onClick={() => onSelect(option)}
+    >
+      <StepProgressDropdownItemContent option={option} />
+    </div>
+  );
+};
+
+StepProgressDropdownItem.propTypes = {
+  option: DROPDOWN_OPTION_PROPTYPE.isRequired,
+  onSelect: PropTypes.func,
+};
+
+const StepProgressDropdown = ({
+  options: optionsProps,
+  value,
+  children,
+  onChange,
+}: {
+  options: StepProgressDropdownOption[];
+  value?: string;
+  children?: ReactNode;
+  onChange?: ({ selectedOption }) => void;
+}): JSX.Element => {
+  const element = useRef(null);
+  const [open, setOpen] = useState(false);
+  const toggleOptions = () => setOpen(s => !s);
+  const [options, setOptions] = useState(optionsProps);
+
+  const getSelectedOption = useCallback(
+    () => (value ? options.find(option => option.value === value) : undefined),
+    [options, value]
+  );
+
+  const [selectedOption, setSelectedOption] = useState(() =>
+    getSelectedOption()
+  );
+
+  const selectedOptionIndex = useMemo(
+    () => options.findIndex(option => option.value === selectedOption?.value),
+    [options, selectedOption]
+  );
+
+  const canMoveNext = useMemo(() => selectedOptionIndex < options.length - 1, [
+    selectedOptionIndex,
+    options,
+  ]);
+
+  const handleDocumentClick = e => {
+    if (element.current && !element.current.contains(e.target)) {
+      setOpen(false);
+    }
+  };
+
+  const handleOptionSelected = useCallback(
+    (newSelectedOption?: StepProgressDropdownOption): void => {
+      if (newSelectedOption?.value === selectedOption?.value) {
+        return;
+      }
+
+      setOpen(false);
+      setSelectedOption(newSelectedOption);
+
+      if (newSelectedOption) {
+        newSelectedOption.activated = true;
+        newSelectedOption.onSelect?.();
+      }
+
+      if (onChange) {
+        onChange({ selectedOption: newSelectedOption });
+      }
+    },
+    [selectedOption, onChange]
+  );
+
+  const handleNextButtonClick = useCallback(() => {
+    if (canMoveNext) {
+      handleOptionSelected(options[selectedOptionIndex + 1]);
+    }
+  }, [options, selectedOptionIndex, canMoveNext, handleOptionSelected]);
+
+  // Update the options in case the options from props has changed
+  useEffect(() => {
+    setOptions(optionsProps);
+    setSelectedOption(getSelectedOption());
+  }, [optionsProps, getSelectedOption]);
+
+  // Updates the selected item based on the value from props
+  useEffect(() => {
+    if (!value) {
+      return;
+    }
+
+    const newOption = value
+      ? options.find(option => option.value === value)
+      : undefined;
+
+    handleOptionSelected(newOption);
+  }, [value, options, handleOptionSelected]);
+
+  const renderOptions = () => {
```

Comment:
have useEffects one after another and don't add logic/component in between them for readability

Reply:
Only another code inherited from `Dropdown.tsx`. Function removed.